### DISCUSSION TITLE
Enabling TLS on selenium grid using custom org managed certificates root ca

### DISCUSSION THREAD

**Discussion by nikhil-rajan-neu** [SeleniumHQ/docker-selenium#2307] [Q&A]
I have trying to enable tls on the selelnium grid , but the current documentation is not very clear to me . 
I have followed this 
helm upgrade -i $RELEASENAME -n $NAMESPACE docker-selenium/selenium-grid \
    --set tls.enabled=true \
    --set-file tls.secretFiles.tls\.crt=/path/to/your/tls.crt \
    --set-file tls.secretFiles.tls\.key=/path/to/your/tls.key \
    --set-file tls.secretFiles.server\.jks=/path/to/your/server.jks
    
   But my node registration is failing with 401

**Comment by VietND96** [comment]
Hi, there is a delay so the new chart version for this doc has not been published yet.
Are you cloning the project and using the source for your trial?

**Comment by nikhil-rajan-neu** [comment]
Yes ..  i got the certificates figured out. But my hub ui is not loading when port forwarding

Sent from Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Reply by VietND96** [reply]
I would understand more about port forwarding. Did you access Hub UI via URL something like `https://your_public_ip/ui/`? If yes, can you try `https://your_public_ip` or `https://your_public_ip/ui`

**Reply by nikhil-rajan-neu** [reply]
@VietND96 yes i am trying to access using the ingress url 
i see that i have this error in my ingress logs 
SSL fake certificate created" file="/etc/ingress-controller/ssl/default-fake-certificate.pem
looks like for some reason its not taking the certificate i am providing

**Reply by VietND96** [reply]
So, how did you deploy the ingress nginx? using its helm chart? If yes, I think you can refer https://github.com/helm/charts/issues/1495
set default-ssl-certificate mount the same selenium TLS secret.
In here I also mentioned: https://github.com/SeleniumHQ/docker-selenium/blob/trunk/charts/selenium-grid/README.md#secure-connection-to-selenium-grid-components

**Reply by nikhil-rajan-neu** [reply]
Yes i am passing that .. here  is my values file 

`basicAuth:
  enabled: true
edgeNode:
  enabled: false
firefoxNode:
  enabled: false
global:
  seleniumGrid:
    defaultNodeStartupProbe: httpGet
hub:
  livenessProbe:
    enabled: false
ingress:
  hostname: selenium.xxx.int
ingress-nginx:
  controller:
    extraArgs:
      default-ssl-certificate: selenium-grid/seleniumgrid-selenium-tls-secret
    service:
      annotations:
        service.beta.kubernetes.io/azure-load-balancer-internal: true
      loadBalancerIP: 10.84.1.112
  enabled: true
tls:
  enabled: true
  ingress:
    enabled: true
`

**Reply by nikhil-rajan-neu** [reply]
@VietND96  .. i cloned whats in trunk and deployed . did not explicitly install nginx .. used what is available in the selenium grid helm chart
